### ISSUE TITLE
Saving sub-scenes will mess up the display of the main scene on the 2D screen

### ISSUE THREAD

**Issue by Rindbee** [godotengine/godot#90301]
### Tested versions

v4.3.dev5.official [c9c17d6ca] good
f47f4a02c bad

### System information

Linux Mint 21.3 (Virginia) - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1050 Ti (nvidia; 535.161.07) - Intel(R) Core(TM) i5-7300HQ CPU @ 2.50GHz (4 Threads)

### Issue description

Saving the sub-scene and switching to the main scene reloads the scene, and it seems like the position calculation of something is dependent on the current camera.

If there is a **enabled** (default value) `Camera2D` node in the main scene, some things in the main scene will be displayed in the wrong position after reloading. For example, the position of the camera screen rectangle. `Parallax2D`'s `screen_offset` may also change.

https://github.com/godotengine/godot/assets/30386067/869cba0c-e815-47b0-a7dc-5fad8c0f11fa

Closing and reopening the scene will display correctly.

### Steps to reproduce

1. Download and use godot to open the MRP;
2. Open main.tscn;
3. Open sub.tscn, and save;
4. Switch to the tab of main.tscn, it will be automatically reloaded, and the displayed content will be messed up after reloading.

### Minimal reproduction project (MRP)

[wrong_display.zip](https://github.com/godotengine/godot/files/14893243/wrong_display.zip)

**Comment by AThousandShips** [comment]
CC @ajreckof @KoBeWi

**Comment by ajreckof** [comment]
it seems the Camera is always rendered at the exact same position and everything else is rendered based on there relative position as if it became the actual Camera of the 2D editor.

**Comment by Rindbee** [comment]
I discovered that the auto-reload here causes `Camera2D::_is_editing_in_editor()` (i.e. `is_part_of_edited_scene()`) to evaluate to `false`. This causes `make_current()` to be called when entering the scene tree.

https://github.com/godotengine/godot/blob/e5b4ef8e9522e950033cbece39a31a4a76da19c1/scene/2d/camera_2d.cpp#L314-L316

https://github.com/godotengine/godot/blob/e5b4ef8e9522e950033cbece39a31a4a76da19c1/scene/main/node.cpp#L2573-L2576

This may be due to the current editing scene root not being set to the main scene root after switching tabs but before auto-reloading.

**Comment by KoBeWi** [comment]
`is_part_of_edited_scene()` *technically* returns true for siblings of edited scene, I think we could check straight for the root viewport instead of using scene root parent.

**Comment by Rindbee** [comment]
It doesn't seem to be fully fixed by #90317. `get_tree()->get_edited_scene_root()` has not changed.

![1](https://github.com/godotengine/godot/assets/30386067/f48cf75e-f245-437e-9492-54e418022982)

It seems that we can not use `repace_by`, because the diffs have been packed before.

https://github.com/godotengine/godot/blob/a7b860250f305f6cbaf61c30f232ff3bbdfdda0b/editor/editor_data.cpp#L704-L711

So directly call `EditorNode::set_edited_scene()` to set `new_scene`, and then delete `old_root` to complete the replacement.

```diff
diff --git a/editor/editor_data.cpp b/editor/editor_data.cpp
index 72225fd454..bdc6504417 100644
--- a/editor/editor_data.cpp
+++ b/editor/editor_data.cpp
@@ -722,8 +722,7 @@ bool EditorData::check_and_update_scene(int p_idx) {
 
                new_scene->set_scene_file_path(edited_scene[p_idx].root->get_scene_file_path());
                Node *old_root = edited_scene[p_idx].root;
-               edited_scene.write[p_idx].root = new_scene;
-               old_root->replace_by(new_scene, false, false);
+               EditorNode::get_singleton()->set_edited_scene(new_scene);
                memdelete(old_root);
                edited_scene.write[p_idx].selection = new_selection;

```
